lib/framework/output.ts

```diff
@@ -125,34 +151,91 @@ async function outputDataItem(
 		}
 	});
 
+	if (
+		(options.displayNullValuesAs || options.hideNullOrUndefinedValues) &&
+		!options.json
+	) {
+		processNullValues(outData, options);
+	}
+
 	if (options.json) {
 		printLine(JSON.stringify(outData, undefined, 2));
 	} else {
-		const chalk = getChalk();
-		const { capitalize } = await import('lodash');
-
 		// Find longest key, so we can align results
 		const longestKeyLength = getLongestObjectKeyLength(outData);
 
+		if (options.titleField) {
+			printTitle(data[options.titleField as keyof any[]], options);
+		}
+
 		// Output one field per line
 		for (const [k, v] of Object.entries(outData)) {
 			const shim = ' '.repeat(longestKeyLength - k.length);
-			const kDisplay = capitalize(k.replace(/_/g, ' '));
-			printLine(`${chalk.bold(kDisplay) + shim} : ${v}`);
+			let kDisplay = k.replace(/_/g, ' ');
+			if (!options.noCapitalizeKeys) {
+				kDisplay = capitalize(kDisplay);
+			}
+			if (k !== options.titleField) {
+				printLine(` ${bold(kDisplay) + shim} : ${v}`);
+			}
 		}
 	}
 }
 
-function getLongestObjectKeyLength(o: any): number {
-	return Object.keys(o).length >= 1
-		? Object.keys(o).reduce((a, b) => {
-				return a.length > b.length ? a : b;
-		  }).length
-		: 0;
+/**
+ * Amend null/undefined values in data as per options:
+ *  - options.displayNullValuesAs will replace the value with the specified string
+ *  - options.hideNullOrUndefinedValues will remove the property from the data
+ *
+ * @param data The data object to process
+ * @param options Output options
+ */
+function processNullValues(data: any, options: InternalOutputOptions) {
+	Object.entries(data).forEach(([k, v]) => {
+		if (v == null) {
+			if (options.displayNullValuesAs) {
+				data[k] = options.displayNullValuesAs;
+			} else if (options.hideNullOrUndefinedValues) {
+				delete data[k];
```

Comment:
Yeah this crossed my mind too, but as you say, we would not expect to use the data after the output logic was called, so I didn't worry about it too much.  But you're right - it would be better to not modify the input.